lib/ddtrace/contrib/grpc/datadog_server_interceptor.rb

```diff
@@ -0,0 +1,54 @@
+require 'grpc'
+require 'ddtrace/propagation/grpc_propagator'
+
+module Datadog
+  module Contrib
+    module GRPC
+      # The DatadogServerIntercptor implements the basic interface
+      # for gRPC middleware. When you declare Datadog usage of the gRPC
+      # module, the interceptor must be explicitly declared for usage.
+      class DatadogServerInterceptor < ::GRPC::ServerInterceptor
+        def request_response(request: nil, call: nil, method: nil)
+          trace_server('server', call.metadata) { yield }
+        end
+
+        def client_streamer(call: nil, method: nil)
+          trace_server('server', call.metadata) { yield }
+        end
+
+        def server_streamer(request: nil, call: nil, method: nil)
+          trace_server('server', call.metadata) { yield }
+        end
+
+        def bidi_streamer(requests: nil, call: nil, method: nil)
+          trace_server('server', call.metadata) { yield }
+        end
+
+        private
+
+        def ddtracer
+          @ddtracer ||= @options.fetch(:ddtracer, Datadog.tracer)
```

Comment:
Just a note that the same general feedback from `DatadogClientInterceptor` likely applies here, too.